router/router.go

```diff
@@ -171,7 +171,7 @@ type Router struct {
 	Shutdown        func()
 }
 
-func New(cfg *config.Configuration, rateConvertor *currency.RateConverter) (r *Router, err error) {
+func New(cfg *config.Configuration, rateConvertor *currency.RateConverter, revision string) (r *Router, err error) {
```

Comment:
Passing down the revision begins a long chain of refactors. I'm personally fine referring back to the static variable, such as `main.Version`.

Reply:
Yeah, wasn't sure here whether to pass it around or access the populated variable... but since main is not importable we would need to move it somewhere else. Should I create a package named version that would contain both these variables?

Reply:
Hi @rokostik we discussed this a bit offline and we support creating a new package put the `version` value in. Maybe inside the `util/version/version.go` directory? We believe it'd greater simplify the code.
```
138             errs = append(errs, &errortypes.FailedToRequestBids{Message: "The adapter failed to generate any bid requests, but also failed to generate an error explaining why"})
139         }
140         return nil, errs
141     }
142
143 -   xPrebidHeader := buildXPrebidHeader(request, version)
    +   xPrebidHeader := buildXPrebidHeader(request, versionValuePackage.GetVersion())
144
145     for i := 0; i < len(reqData); i++ {
146         if reqData[i].Headers != nil {
147             reqData[i].Headers = reqData[i].Headers.Clone()
exchange/bidder.go
```
What do you think? (Feel free to name the package to something better than `versionValuePackage`)

Reply:
> but since main is not importable

Doh! I didn't think of that.

> Maybe inside the `util/version`

Maybe a version package at the top level? rather than within util? Doesn't really matter. So long as it's easily discoverable.

Reply:
I think a top level version directory would work best in this case. I'll also move the Rev variable to there.

Reply:
Extracted both Rev and Ver (newly named Version) to their own package. I also renamed the /version endpoint to /revision and added a new /version endpoint. So that these two endpoints actually return the things they're named after.

Reply:
> Extracted both Rev and Ver (newly named Version) to their own package.

Perfect.

>  I also renamed the /version endpoint to /revision and added a new /version endpoint. So that these two endpoints actually return the things they're named after.

This is technically a breaking change, but it makes sense and should be fine since it's currently broken. I don't believe it's necessary to add an new revision endpoint, we could keep the version endpoint and change the response to include both:

```
type versionModel struct {
	Version string `json:"version"`
	Revision string `json:"revision"`
}
```

I also don't know why my predecessors on this project choose to use a short hash instead of the full hash. What do you think of changing the build script to use the full hash for Rev?

Reply:
Good idea, I now merged both under the /version endpoint. 
As for why it's the short sha, I couldn't find the reason in the original issue and since you say that it's not used anywhere I think it's ok if I change it to the full hash so I went ahead and did that.